neural_modelling/src/neuron/synapses.c

```diff
@@ -202,29 +202,48 @@ static inline bool process_fixed_synapses(
         // Get the next 32 bit word from the synaptic_row
         // (should auto increment pointer in single instruction)
         uint32_t synaptic_word = *synaptic_words++;
+        uint32_t synapse_type = synapse_row_sparse_type(
+            synaptic_word, synapse_index_bits, synapse_type_mask);
+
+        // If synapse type has non-input synapses and this synapse
+        // connects to one, pass event directly to synapse dynamics
+        if (synapse_type > 1) {
+            // Dopaminergic neurons send some amount of neuromodulator
+            // concentration so this can actually be a weight as usual.
+            int32_t concentration = synapse_row_sparse_weight(synaptic_word);
+            uint32_t index = synapse_row_sparse_index(
+                synaptic_word, synapse_index_mask);
+            // In case this is punishment synapse, invert dopamine level
+            // to cause depression.
+            if (synapse_type == 3) {
+                concentration = ~concentration + 1;
+            }
+            synapse_dynamics_process_neuromodulator_event(time,
+                concentration, index, synapse_type);
+        } else {
```

Comment:
I guess again here that there isn't an example using a different synapse type where the ID could be > 1 (though I think the dual-excitatory synapse is the only current one we have?) along with neuromodulation, so it wouldn't have been noticed yet.  I agree though that this could be moved into an impl file somehow.

Reply:
I would think this would already break the dual-excitatory synapse... it might be that we don't have a test for this that actually checks the result (it wouldn't fail to run, but it would likely break the results).  I think EProp might also make use of the multiple synapse types though I am not certain...